### PR TITLE
test: Edge cases - Consensus Service Tests

### LINKED ISSUE #19758: Edge cases - Consensus Service Tests

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
* Add end-to-end business cases for topics wrapped in an atomic batch

**Related issue(s)**:

Fixes #19758

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/AtomicSubmitMessageSuite.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+import com.hedera.services.bdd.spec.keys.SigControl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DynamicTest;
+
+// This test cases are direct copies of SubmitMessageSuite. The difference here is that
+// we are wrapping the operations in an atomic batch to confirm the fees are the same
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/AtomicSubmitMessageSuite.java]: Looks like a copy-paste error–very few fees are actually checked in all of these classes

**Review by mhess-swl — COMMENTED** [review]
I'm fine with merging these tests (once the comment about fees for each suite is fixed), but with a caveat that these tests alone should not close the ticket for consensus end-to-end business cases (#19390). Wrapping the existing hapi tests and resubmitting them as batch transactions doesn't cover any end-to-end scenarios not previously thought of; this is most apparent in that each test suite is focused only on a single transaction type, e.g. `SubmitMessageSuite`, `TopicCreateSuite`. For business end-to-end cases, we want tests that exercise all parts of the entity's lifecycle–most likely _within the same test_. For example, what happens if a topic is created, submitted to three times, has an admin key changed, and then is deleted, all within a single batch transaction? These types of scenarios may expose a different class of bugs.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/AtomicSubmitMessageSuite.java (hunk 1 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/AtomicSubmitMessageSuite.java]: Fixed 👍

**Comment by vtronkov** [comment]
> I'm fine with merging these tests (once the comment about fees for each suite is fixed), but with a caveat that these tests alone should not close the ticket for consensus end-to-end business cases (#19390). Wrapping the existing hapi tests and resubmitting them as batch transactions doesn't cover any end-to-end scenarios not previously thought of; this is most apparent in that each test suite is focused only on a single transaction type, e.g. `SubmitMessageSuite`, `TopicCreateSuite`. For business end-to-end cases, we want tests that exercise all parts of the entity's lifecycle–most likely _within the same test_. For example, what happens if a topic is created, submitted to three times, has an admin key changed, and then is deleted, all within a single batch transaction? These types of scenarios may expose a different class of bugs.

My thought process was that we've already had edge case tests for the consensus, and wrapping them into an atomic batch would give us good coverage. However, I agree that these tests are not end-to-end 👍 

Just note that these types of scenarios are not supported inside a batch:
```
...topic is created, submitted to three times, has an admin key changed, and then is deleted, all within a single batch transaction
```
- We cannot create an entity inside a batch and then use it within the same batch. There is no way to get the entityID to call it. And this behavior is not defined in the HIP itself.
- The change of a key within a batch will be supported after Jivko's PR is done and merged - https://github.com/hiero-ledger/hiero-consensus-node/pull/19731

Even though these are not end-to-end they are edge cases worth testing so what I did was unassign myself from the end-to-end issues, created a new subissue for the edge cases https://github.com/hiero-ledger/hiero-consensus-node/issues/19756, and mapped my PRs to their corresponding tickets.

**Comment by mhess-swl** [comment]
> We cannot create an entity inside a batch and then use it within the same batch. There is no way to get the entityID to call it. And this behavior is not defined in the HIP itself.

Copy that, I learned this the hard way 😅😭😂 

> Even though these are not end-to-end they are edge cases worth testing so what I did was unassign myself from the end-to-end issues, created a new subissue for the edge cases https://github.com/hiero-ledger/hiero-consensus-node/issues/19756, and mapped my PRs to their corresponding tickets.

This sounds great, thank you!!

**Review by mhess-swl — APPROVED** [review]
